shared/src/models/user.model.ts

```diff
@@ -0,0 +1,8 @@
+import {ResourceWithoutId} from './resource.model';
+export interface IUser extends ResourceWithoutId {
```

Comment:
extending interface with index signature is extremely dangerouse - typos are not detectable:
```ts
const user: IUser = {} as IUser;
user.emails = 2;
```

Reply:
Okay. What's the best alternative?

Reply:
Just drop the index signature - it's purpose are dictionaries, object map. Just leave only the `id` property in `Resource`, the rest of models will extends it and have only the declared properties.

Reply:
ok